### ISSUE TITLE
Support for AKS API Server VNet Integration

### ISSUE THREAD

**Issue by zioproto** [Azure/terraform-azurerm-aks#393]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

AKS API Server VNet Integration. The product is still in preview but let's track it in a GitHub issue so we are ready to merge a PR as soon is promoted to GA.

Resources:
* https://github.com/Azure/AKS/issues/2729
* https://learn.microsoft.com/en-us/azure/aks/api-server-vnet-integration
* https://www.youtube.com/watch?v=54y986U1uYM&t=494s

Status:
* Product still in preview.
* Implementation in the Terraform provider is available. https://registry.terraform.io/providers/hashicorp/azurerm/3.61.0/docs/resources/kubernetes_cluster#vnet_integration_enabled

### New or Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Potential Terraform Configuration

```hcl
variable "vnet_integration_enabled" {
  type        = bool
  default     = false
  description = "(Optional) Should API Server VNet Integration be enabled? For more details please visit Use API Server VNet Integration."
}

resource "azurerm_kubernetes_cluster" "main" {
   [..CUT..]
   vnet_integration_enabled = var.vnet_integration_enabled
   [..CUT..]

}
```

**Comment by richshadman** [comment]
Any chance of this possibly getting this reprioritized and reclassified?

I am trying to create a cluster that is entirely within our private network and would love to make use of this feature. Right now trying to set the subnet results in the following error:
![image](https://github.com/Azure/terraform-azurerm-aks/assets/59030423/6c09e874-6caa-4d3c-bc1b-fe188a59c688)

I would argue that this is actually a **bug** in the current version of the module as the module allows setting a subnet here: 
![image](https://github.com/Azure/terraform-azurerm-aks/assets/59030423/dbfa0bbb-cb30-4c0b-816d-2c40cce1d483)

I think instead of adding a new variable, we should simply set this to true if the subnet is passed.

@lonegunmanb would it be possible to reclassify as a bug and make the suggested change?

**Comment by ryan-grenz-evelyn** [comment]
@richshadman I am hitting the same problem now, though my problem is because of KMS keyvault network access being set to Private. I'm pretty sure a week or so ago when I last deployed the cluster with the same settings the Azure API didn't enforce the vnet integration requirement, but it certainly does now. 

Anyway, I would really like to have this switch available and it looks like the @bislydev commit above would add it?

**Comment by zioproto** [comment]
@richshadman you are right. I identified a mistake on our side, we merged the `var.api_server_subnet_id` variable that is actually part of the preview feature API Server VNET Intergration, in PR https://github.com/Azure/terraform-azurerm-aks/pull/381

Please bare with us until the feature is declared as GA. Then me and @lonegunmanb will update this module as soon as possible.

**Comment by zekena2** [comment]
Any info when will it be GA?

**Comment by ddezoysa** [comment]
@richshadman @lonegunmanb @zioproto When this will be in GA? Please help.

**Comment by joemiao-exos** [comment]
Preview features should be included in aks module. It's been really painful not being able to manage some preview features but others. 

For example,

I have to run the following command to enable API Server Vnet Integration outside of the cluster. 

`az aks update --name xxx --resource-group xxx --enable-apiserver-vnet-integration --apiserver-subnet-id "xxxx"`

The real pain comes when I run a tf plan on the aks module after ANY change, it shows a difference in TF and remote infra.
And I have no way to ignore this change as vnet_integration_enabled is not a supported input in this aks module. 
I have no way around it. 

```
      ~ api_server_access_profile {
          - vnet_integration_enabled = true -> null
            # (2 unchanged attributes hidden)
        }
```

**Comment by aamoctz** [comment]
@lonegunmanb this feature is now GA 
https://techcommunity.microsoft.com/blog/AppsonAzureBlog/build-secure-flexible-ai-enabled-applications-with-azure-kubernetes-service/4415227

**Comment by amoosbr** [comment]
Hi,

terraform-provider-azurerm  [v4.46.0](https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v4.46.0) released yesterday with support for API server virtual network integration :

> azurerm_kubernetes_cluster - add support for the api_server_access_profile.virtual_network_integration_enabled and api_server_access_profile.subnet_id properties (https://github.com/hashicorp/terraform-provider-azurerm/issues/30559)

I think, the issue is now no longer blocked.

**Comment by ryan-grenz-evelyn** [comment]
I honestly don't think you're going to see much more from this repo - it hasn't been updated since September. This module does support it though: https://github.com/Azure/terraform-azurerm-avm-res-containerservice-managedcluster

<img width="1327" height="682" alt="Image" src="https://github.com/user-attachments/assets/f8be45f0-0a77-4609-bbe5-8c15debe2a32" />

**Comment by monotek** [comment]
Imho, this has been implemented for a while and this issue can be closed.

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#api_server_access_profile-1
